lib/store.js

```diff
@@ -115,7 +115,7 @@ Store.prototype.get = function (packet, cb) {
  */
 Store.prototype.close = function (cb) {
   if (this.options.clean) {
-    this._inflights = null
+    this._inflights = {}
```

Comment:
I don't think this is generically correct. I think _we should not close `store.close()`  in this scenario.

Reply:
I agree with you. `Store.close()` is not an appropriate point to prepare `reconnect()`.

When I revert `this._inflights = null`, the tests I added don't pass.
https://github.com/mqttjs/MQTT.js/pull/734/files#diff-16702ac352d8772fe21f4f9c1de7e900R2031

The connection is `clean: true` (default). In this case, I think that user should be able to call `client.end()` and then call `client.reconnect()` in the close handler.
However, call `client.reconnect()` or not is unpredictable, so I set `_inflights` to `{}` to prepare next `client.reconnect()` calling.

I think that `store.close()` should be called because `client.end()` is called because of https://github.com/mqttjs/MQTT.js/pull/707#discussion_r148088223

Finally, I found the appropriate point to reset `Store._inflights`. It is `reconnect()` function. I renew the `incomingStore` and `outgoingStore` there. They could be user custom object so I call `constructor` explicitly.

Please check out my next commit.